### PR TITLE
Added Mindwarper

### PR DESCRIPTION
**Pull request by starryburst**
A role that is similar to cthulu, but can drive people insane more consistently and allows for more counterplay.

### CODE REVIEW

File: Games/types/Mafia/roles/cards/MindShifter.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+                    if (!this.actor.role.data.insane){
+                        return;
+                    }
+
+                    var stillInsane = true;
+                    var visitors = this.getVisitors(this.actor.role.data.insane);
+                    for (let visitor of visitors){
+                        if (visitor.role.alignment != "Monsters")
+                            stillInsane = false;
```
Comment (pikulet) [inline Games/types/Mafia/roles/cards/MindShifter.js:41]: might i suggest adding a "break" here

File: data/roles.js [hunk 2]
```diff
@@ -961,6 +961,13 @@ const roleData = {
                 "Once a potion has been concocted, it cannot be brewed again for the next two nights."
             ],
         },
+        "Mindwarper": {
+            alignment: "Monsters",
+            description: [
+                "Visits a player each night.",
+                "If that player is not visited by a non-Monsters player during the next night, they will go insane."
```
Comment (pikulet) [inline data/roles.js:968]: Monster should be singular here

File: Games/types/Mafia/roles/cards/MindShifter.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+                            break;
+                        }
+
+                    if (stillInsane){
```
Comment (Fsalker) [inline Games/types/Mafia/roles/cards/MindShifter.js:45]:
```suggestion
                    const becomesInsane = !visitors.find(visitor => visitor.role.alignment != "Monsters")

                    if (becomesInsane){
```

javascript for loops are banned in 52 states

always use `let`/`const`. never use `var` unless you know you have to (or you learnt javascript in 1999)

https://dev.to/paritho/3-reasons-to-use-var-in-javascript-1hoe
https://softwareengineering.stackexchange.com/questions/274342/is-there-any-reason-to-use-the-var-keyword-in-es6

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array#instance_methods

File: Games/types/Mafia/roles/cards/MindShifter.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+            "Warp Mind": {
+                states: ["Night"],
+                flags: ["voting"],
+                targets: { include: ["alive"], exclude: ["Monsters"] },
+                action: {
+                    labels: ["effect"],
+                    priority: PRIORITY_EFFECT_GIVER_DEFAULT,
+                    run: function () {
+                        this.actor.role.data.insane = this.target;
+                        this.target.queueAlert("You will be driven insane if not visited by a player not aligned with the Monsters, tonight!");
```
Comment (pikulet) [inline Games/types/Mafia/roles/cards/MindShifter.js:19]: if it's tonight then the priority is wrong, should be +1

Comment (starryburst) [reply Games/types/Mafia/roles/cards/MindShifter.js:19]: the player receives this message the follwing day

timeline would look like:

N0 - picks person
D1 - "you will be driven insane if not visited tonight"
N1 - action runs during night
D2 - determines if player becomes insane or not

Comment (pikulet) [reply Games/types/Mafia/roles/cards/MindShifter.js:19]: I see, the action is meant for the next night. Thanks!

**Review by pikulet — APPROVED** [review]

**Pull request merged**